### PR TITLE
Introduced `metadata.osio.enabled: false` where necessary

### LINKED ISSUE #242: Introduce a specific metadata entry for boosters enabled in OSIO
Making OSIO as a `runsOn` value is becoming problematic with new boosters. We decided that the best is for the Booster metadata to have a `osio.enabled: true` flag on boosters that should appear in OSIO and change the frontend code to query on that instead

### PR DESCRIPTION
**Pull request by quintesse**
Removed mention of `osio` from all `metadata.runsOn` properties and
used `metadata.osio.enabled` instead.

Fixes #242

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,13 +73,13 @@ Path   | Description
 ------ | -----------
 runsOn | (Optional) A single cluster type or a list of cluster types where this booster can run. A type can be prefixed with `!` to negate the option: the booster will _not_ run on clusters of that type. Special values `all` and `none` indicate the booster will run everywhere or nowhere respectively. Important: when using the `!` to negate you _must_ surround it and the cluster type name with double quotes.
 buildProfile | (Optional) The Maven profile that should be activated in the booster's `pom.xml` file
+osio.enabled | (Optional) Determines if the booster should be shown in OSIO (defaults to `true`)
```
[the remark this answers is not in the document]
Comment (quintesse) [reply README.md]: > Maybe use osio/enabled

I actually thought I had used `.` in the rest of the text, but I didn't, in fact I had used `/` before. Good catch. Fixed.

> I am not entirely sure it should default to true

Leaving this as-is for now, as discussed on Slack.

**Comment by quintesse** [comment]
Repushed. Forgot to set `runsOn: none` in the one booster that's strictly for OSIO only

**Comment by gastaldi** [comment]
`runsOn: none` still allows to download as a ZIP, can we avoid that for that specific booster?

File: README.md [hunk 2]
```diff
@@ -73,13 +73,13 @@ Path   | Description
 ------ | -----------
 runsOn | (Optional) A single cluster type or a list of cluster types where this booster can run. A type can be prefixed with `!` to negate the option: the booster will _not_ run on clusters of that type. Special values `all` and `none` indicate the booster will run everywhere or nowhere respectively. Important: when using the `!` to negate you _must_ surround it and the cluster type name with double quotes.
 buildProfile | (Optional) The Maven profile that should be activated in the booster's `pom.xml` file
+osio/enabled | (Optional) Determines if the booster should be shown in OSIO (defaults to `true`)
```
Comment (gastaldi) [inline README.md]: Change to `app/osio/enabled` and add `app/launcher` too

File: README.md [hunk 3]
```diff
@@ -73,13 +73,13 @@ Path   | Description
 ------ | -----------
 runsOn | (Optional) A single cluster type or a list of cluster types where this booster can run. A type can be prefixed with `!` to negate the option: the booster will _not_ run on clusters of that type. Special values `all` and `none` indicate the booster will run everywhere or nowhere respectively. Important: when using the `!` to negate you _must_ surround it and the cluster type name with double quotes.
```
Comment (gastaldi) [inline README.md]: Change to `app/launcher/runsOn`

Comment (quintesse) [reply README.md]: Yes yes ;)

**Comment by quintesse** [comment]
*sigh* Exactly why I had wanted this merges as soon as possible :(
https://github.com/fabric8-launcher/launcher-booster-catalog/pull/245

**Pull request merged**